### DISCUSSION TITLE
RFC - App-wide shared static/stale data

### DISCUSSION THREAD

**Discussion by Vadorequest** [UnlyEd/next-right-now#335] [Show and tell]
I've implemented https://github.com/UnlyEd/next-right-now/pull/334 today

![image](https://user-images.githubusercontent.com/3807458/119705146-b3ed3980-be58-11eb-9c6f-0ff0476bdb59.png)

It has a few benefits, especially around build optimizations, and also only generates the necessary pages (for actives languages) instead of generating all pages for all languages no matter what.

It also has a few considerations, I've found the plugin to make the app much less resilient to changes, and provide quite a bad dev experience on some troubling code issues, see https://github.com/ricokahler/next-plugin-preval/issues/26

Overall, I like the changes, but I'm concerned about maintainability. It's an optimization I've wanted to do for more than a year, I hope the `next-plugin-preval` improves and fixes those developer-facing issues.

Don't hesitate to share your thoughts!

**Comment by Vadorequest** [comment]
Similar optimization has been implemented for i18n (Locize) too: https://github.com/UnlyEd/next-right-now/pull/337

**Reply by samuelcastro** [reply]
This is really interesting. I wasn't aware about this plugin.

**Comment by Vadorequest** [comment]
I've encountered more issues, especially with larger datasets (our own private app):
- https://github.com/ricokahler/next-plugin-preval/issues/27
- ~~Another issue due to circular references, build on Vercel fails (the NRN demo doesn't have circular data, that's why it wasn't affected)~~ My own mistake trying to JSON.stringify circular data

I'll disable the plugin during development, it's a shame but I don't see a better alternative until it's fixed.

**Reply by Vadorequest** [reply]
PR at https://github.com/UnlyEd/next-right-now/pull/341
